File: src/ServiceBusQueueTrigger.cs
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;

namespace FunctionApp;

public class ServiceBusQueueTrigger
{
    private readonly ILogger<ServiceBusQueueTrigger> _logger;

    public ServiceBusQueueTrigger(ILogger<ServiceBusQueueTrigger> logger)
    {
        _logger = logger;
    }

    [Function(nameof(ServiceBusQueueTrigger))]
    public async Task Run(
        [ServiceBusTrigger("%ServiceBusQueueName%", Connection = "ServiceBusConnection")] string message)
    {
        _logger.LogInformation("C# ServiceBus Queue trigger start processing a message: {Message}", message);
        await Task.Delay(5000);
        _logger.LogInformation("C# ServiceBus Queue trigger end processing a message");
    }
}

File: src-ts/src/functions/serviceBusQueueTrigger.ts
import { InvocationContext } from '@azure/functions';

export async function serviceBusQueueTrigger(message: unknown, context: InvocationContext): Promise<void> {
    const messageBody = typeof message === 'string' ? message : JSON.stringify(message);
    
    context.log('TypeScript ServiceBus Queue trigger start processing a message:', messageBody);
    
    // Simulate processing work with a 5-second delay
    await new Promise(resolve => setTimeout(resolve, 5000));
    
    context.log('TypeScript ServiceBus Queue trigger end processing a message');
}
